NavigationReactNative/src/android/app/src/main/java/com/navigation/reactnative/NavigationBarManager.java

```diff
@@ -0,0 +1,124 @@
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.ViewGroupManager;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+
+public class NavigationBarManager extends ViewGroupManager<NavigationBarView> {
+    @Nonnull
+    @Override
+    public String getName() {
+        return "NVNavigationBar";
+    }
+
+    @Nonnull
+    @Override
+    protected NavigationBarView createViewInstance(@Nonnull ThemedReactContext reactContext) {
+        return new NavigationBarView(reactContext);
+    }
+
+    @Override
+    public int getChildCount(NavigationBarView parent) {
+        return parent.toolbar.getChildCount();
+    }
+
+    @Override
+    public View getChildAt(NavigationBarView parent, int index) {
+        return parent.toolbar.getChildAt(index);
+    }
+
+    @Override
+    public void addView(NavigationBarView parent, View child, int index) {
+        Toolbar.LayoutParams layoutParams = new Toolbar.LayoutParams(
+                Toolbar.LayoutParams.MATCH_PARENT,
+                Toolbar.LayoutParams.MATCH_PARENT
+        );
+        parent.toolbar.addView(child, index, layoutParams);
+        parent.toolbar.setContentInsetsRelative(0, 0);
+    }
+
+    @Override
+    public void removeViewAt(NavigationBarView parent, int index) {
+        parent.toolbar.removeViewAt(index);
+        parent.toolbar.setContentInsetsRelative(parent.defaultContentInsetStart, parent.defaultContentInsetEnd);
+    }
+
+    @ReactProp(name = "title")
+    public void setTitle(NavigationBarView view, @Nullable String title) {
+        view.setTitle(title);
+    }
+
+    @ReactProp(name = "logo")
+    public void setLogo(NavigationBarView view, @Nullable ReadableMap logo) {
+        view.setLogoSource(logo);
+    }
+
+    @ReactProp(name = "navigationImage")
+    public void setNavIcon(NavigationBarView view, @Nullable ReadableMap navIcon) {
+        view.setNavIconSource(navIcon);
+    }
+
+    @ReactProp(name = "overflowImage")
+    public void setOverflowIcon(NavigationBarView view, @Nullable ReadableMap overflowIcon) {
+        view.setOverflowIconSource(overflowIcon);
+    }
+
+    @ReactProp(name = "menuItems")
+    public void setMenuItems(NavigationBarView view, @Nullable ReadableArray menuItems) {
+        view.setMenuItems(menuItems);
+    }
+
+    @ReactProp(name = "barTintColor", customType = "Color")
+    public void setBarTintColor(NavigationBarView view, @Nullable Integer barTintColor) {
+        if (barTintColor != null) {
+            view.setBackground(null);
+            view.toolbar.setBackgroundColor(barTintColor);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && Color.alpha(barTintColor) < 255)
+                view.setOutlineProvider(null);
```

Comment:
We need to restore the default outline provider if the color is set and not transparent. We might be changing from a transparent to a non-transparent color after a `setState`